cpp/include/cpu_operations.h

```diff
@@ -162,24 +162,28 @@ static Vector<T> Norm(const Matrix<T> &a,
     int num_rows = a.rows();
     int num_cols = a.cols();
     float nval = 0;
-    Vector<T> norm(num_cols);
     if (axis == 0) {
+    Vector<T> norm(num_cols);
      for (int j = 0; j < num_cols; j++) {
       for (int i = 0; i < num_rows; i++)
          nval += pow(a(i, j), p);
        norm(j) = pow(nval, (1.0/p));
        nval = 0;
      }
      return norm;
-     } else {
+    } else if (axis == 1) {
+     Vector<T> norm(num_rows);
      for (int i = 0; i < num_rows; i++) {
       for (int j = 0; j < num_cols; j++)
          nval += pow(a(i, j), p);
        norm(i) = pow(nval, (1.0/p));
        nval = 0;
```

Comment:
Not necessarily. I couldn't think of a good name for that variable but I'll try to change it to something clearer.